test/unit/math/rev/mat/fun/unit_vector_transform_test.cpp

```diff
@@ -49,3 +50,12 @@ TEST(probTransform,unit_vector_jacobian) {
     << "x = " << x.transpose();
 }
 
+TEST(probTransform, check_varis_on_stack) {
+  using stan::math::to_var;
+  using stan::math::var;
+  Matrix<var,Dynamic,1> y(3);
+  y << 2, 3, -1;
+  
+  var lp(0);
+  test::check_varis_on_stack(stan::math::unit_vector_constrain(y, lp));
```

Comment:
not sure why this is in a different file than the last test as they're of the same function